api/dao/base.py

```diff
@@ -80,12 +80,17 @@ def get_container(self, _id, projection=None, get_children=False):
             cont[CHILD_MAP[self.cont_name]] = children
         return cont
 
-    def get_children(self, _id, projection=None):
+    def get_children(self, _id, projection=None, uid=None):
         try:
             child_name = CHILD_MAP[self.cont_name]
         except KeyError:
             raise APINotFoundException('Children cannot be listed from the {0} level'.format(self.cont_name))
-        query = {self.cont_name[:-1]: bson.ObjectId(_id)}
+        if not self.use_object_id:
+            query = {containerutil.singularize(self.cont_name): _id}
+            if uid:
+                query['permissions'] = {'$elemMatch': {'_id': uid}}
+        else:
+            query = {self.cont_name[:-1]: bson.ObjectId(_id)}
```

Comment:
And wouldn't you want to respect the uid match no matter the container? I must have missed that in the first review.